### PR TITLE
mm/mm_heap: Add mm_get_heap_info_with_index and Move mm_get_heap_info…

### PR DESCRIPTION
**Pull request by jeongchanKim**
… to mm_get_heap_info.c

1. when getting heap information, user can use allocated address and heap index also.
2. change file name : mm_get_heapindex to mm_get_heap, and integrate funtions related to heap information

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [3db13589c5b735987b1df5ada68ecd00e381ba66] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3db13589c5b735987b1df5ada68ecd00e381ba66] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [434e21ee5130afea735244785e5ada1ba83e3460] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [434e21ee5130afea735244785e5ada1ba83e3460] - Code Rule Check OK.

File: apps/examples/slsiwifi/slsiwifi_main.c [hunk 1]
```diff
@@ -104,7 +105,7 @@ static struct mm_heap_s *g_user_heap;
 int g_memstat_total = 0;
 static int getMemUsage(void)
 {
-	g_user_heap = mm_get_heap_info();
+	g_user_heap = mm_get_heap_info_with_index(BASE_HEAP_INDEX);
```
Comment (yashwanth686007) [inline apps/examples/slsiwifi/slsiwifi_main.c]: In this case, how one can be sure that only base index heap is being used?

Comment (jeongchanKim) [reply apps/examples/slsiwifi/slsiwifi_main.c]: This sample is tested based on single heap. So base heap is used.

**Comment by seinfra** [comment]
Target : [f522920b54addaa3e8e74adeabbcaa664887214e] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [f522920b54addaa3e8e74adeabbcaa664887214e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4944893086c98d484eea0ea453a54e704b715cd6] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4944893086c98d484eea0ea453a54e704b715cd6] - Code Rule Check OK.

**Pull request merged**